### DISCUSSION TITLE
Regex capabilities

### DISCUSSION THREAD

**Discussion by God-damnit-all** [nushell/nushell#4266] [Q&A]
The documentation doesn't cover what type of RegEx nushell's implementation is most similar to. I'm also curious if capture groups support expressions like `\L$1` to force the string to be all lowercase.

**Comment by kubouch** [comment]
I think we use just the regex crate (at least in the `match` command). I haven't worked on regex stuff so I don't have a precise answer.

To convert string to lowercase, you can use `str downcase`.

It would be a good thing to document. I made an issue for it: https://github.com/nushell/nushell/issues/4268 . We'll gladly accept a PR that adds that.

**Reply by God-damnit-all** [reply]
> To convert string to lowercase, you can use `str downcase`.

I phrased that in a weird way, I was more wanting to know if capture group manipulation flags like that were supported or not.

**Reply by kubouch** [reply]
I see. I have no idea since I don't use regex very much. I'd suggest checking the "regex" crate. At least the `match` command uses it. You can check the source code of other commands (it's easy to find under `crates/nu-command`) and see what they use. I'd say chances are we just use the "regex" crate everywhere. We don't have our own regex engine.
